src/Orleans/Configuration/GlobalConfiguration.cs

```diff
@@ -415,7 +460,14 @@ internal GlobalConfiguration()
             ExpectedClusterSizeConfigValue = new ConfigValue<int>(DEFAULT_LIVENESS_EXPECTED_CLUSTER_SIZE, true);
             ServiceId = Guid.Empty;
             DeploymentId = Environment.UserName;
-            DataConnectionString = "";
+
+            DataConnectionStringForLiveness = "";
+            // Assume the ado invariant is for sql server storage if not explicitly specified
+            AdoInvariantForLiveness = AdoNetInvariants.InvariantNameSqlServer;
```

Comment:
I have another suggestion, we can check the app.config and see if the user has added an ado invariant there. In most cases, there will be only one and that's the one the user probably wants.
this allows us to infer the provider automatically and keep any current configuration xml exactly the same as before.
So It can be something like:
* if ado invariant is given in the provider, use it
 * else, if there's one in app.config, use it
     * else, use hard coded System.Data.SqlClient